File: Data Structures & Algorithms/daily-temperatures/submission-2.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        if (temperatures.length == 1) return 0
        let res = new Array(temperatures.length).fill(0)

        let i = 0
        let j = i
        while (i < temperatures.length) {
            j = i + 1
            let count = 1

            while (j < temperatures.length && temperatures[j] <= temperatures[i]) {
                count++
                j++
            }

            if (j >= temperatures.length) {
                count = 0
            }

            res[i] = count
            i++
        }
        
        return res
    }
}

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        a = {}
        for char in s:
            if char not in a:
                a[char] = 1
            else:
                a[char] += 1

        b = {}
        for char in t:
            if char not in b:
                b[char] = 1
            else:
                b[char] += 1

        for char, count in a.items():
            if char not in b:
                return False
            
            if count != b[char]:
                return False

        return True